apps/core/faviconmanager.cpp

```diff
@@ -193,12 +195,14 @@ void FaviconManager::grabIcon(const QString &type, DeclarativeWebPage *webPage,
         QObject::disconnect(*dataConn);
         if (dataFetcher->hasAcceptedTouchIcon()) {
             qCDebug(lcFavoritesLog) << "Storing favicon for" << type;
+qWarning() << "XXXXXXXXXXXXXXX got favicon result: " << dataFetcher->data() << " for url: " << webPage->url().toString() << " with type: " << type;
             add(type, webPage->url().toString(), dataFetcher->data(), true);
         }
         else {
             std::shared_ptr<QMetaObject::Connection> thumbConn = std::make_shared<QMetaObject::Connection>();
             *thumbConn = connect(webPage, &DeclarativeWebPage::thumbnailResult, [this, type, webPage, thumbConn](const QString &data) {
                 qCDebug(lcFavoritesLog) << "Storing thumbnail for" << type;
+qWarning() << "XXXXXXXXXXXXXXX got thumbnail result: " << data << " for url: " << webPage->url().toString() << " with type: " << type;
```

Comment:
instead, the result is always a thumbnail, which happens to be some ... completely nondescript square from somewhere in the page, rather than something useful.  Weird.

Reply:
That's the default fallback in case there is no favicon. I think it takes a square from the top left corner of the page. It would be good to improve on that.

Reply:
True.  I mean, I don't have any better suggestions (maybe choose the most common non-white/black colour from the page?  but that might be expensive to determine...)

Reply:
Yeah, it's tricky. I was thinking maybe grabbing a larger portion of the page and scaling it down. I'm not sure I'd trust myself to write a more complex heuristic based on page contents, but if you think something like that can work, I'm all for it.

Reply:
A scaled down version would probably be far better, if it doesn't get totally mushed due to the low res.  Anyway, it's unrelated to this PR I guess.